app/forms/providers/means/housing_benefit_form.rb

```diff
@@ -1,52 +1,12 @@
 module Providers
   module Means
-    class HousingBenefitForm
-      include ActiveModel::Model
+    class HousingBenefitForm < RegularTransactionForm
+      attr_accessor :housing_benefit_amount, :housing_benefit_frequency
 
-      attr_accessor :housing_benefit,
-                    :housing_benefit_amount,
-                    :housing_benefit_frequency
+      delegate :applicant_in_receipt_of_housing_benefit, to: :legal_aid_application
 
-      attr_reader :legal_aid_application
-
-      validates :housing_benefit, inclusion: { in: %w[true false] }
-
-      validates :housing_benefit_amount,
-                currency: { greater_than: 0 },
-                if: :applicant_in_receipt_of_housing_benefit?
-
-      validates :housing_benefit_frequency,
-                inclusion: { in: ->(form) { form.frequency_options } },
-                if: :applicant_in_receipt_of_housing_benefit?
```

Comment:
Is the conditional `if` logic on these validations accounted for in the superclass or elsewhere?

Reply:
It's covered by `validate :all_regular_transactions_valid`.

If "No" is selected, there are no `transaction_type_ids`, and no regular transactions to validate.

Does that make sense?